tests/rules/defer-persistent-binding.py

```diff
@@ -1,10 +1,14 @@
 
+# OLD BEHAVIOR:
```

Comment:
Previously we thought this was important, but now it's not important anymore? 🤔 What motivated this test that now we're fine breaking? I'm not questioning that the new behavior is useful, but I'm questioning whether we can just stop supporting the old behavior.

Reply:
I doubt anyone was hinging on the old behavior. I don't think this particular implementation was important, I think it was just important that the bindings persisted whatsoever, irrespective of how the bindings were then unified.

I believe this to have been an oversight at the time I implemented persistent bindings. The new behavior is such that you should still get the finding, unless the persisted bindings are in conflict.

Reply:
It may be good to then note here that "previous behavior was <this> but in retrospective this was a mistake for <these reasons>, we prefer <this> new behavior". When I read these comments it felt kinda arbitrary why we changed the behavior, and the old comment here tries to make a point that it's important to produce a finding.